corehq/apps/case_search/utils.py

```diff
@@ -205,10 +207,23 @@ def _apply_filter(self, search_es, criteria):
         elif criteria.key == COMMCARE_PROJECT:
             if not criteria.is_empty:
                 return search_es.filter(filters.domain(criteria.value))
+        elif criteria.key == CASE_SEARCH_SORT_KEY:
+            return search_es.sort_by_case_properties(self._parse_custom_sort_properties(criteria.value))
         elif criteria.key not in UNSEARCHABLE_KEYS:
             return search_es.add_query(self._get_case_property_query(criteria), queries.MUST)
         return search_es
 
+    def _parse_custom_sort_properties(self, sort_properties):
```

Comment:
I think that could make sense with the perspective of sorting taking place before search filters are applied, rather than being just another filter. It also seems like it could allow more flexibility in things like allowing using `commcare_sort` filters from both this UI and the default search filters, though I'm not sure why you'd do that.

Would a version of `_parse_custom_sort_properties` then just be used as a `converter`, something like this?
```python
    commcare_sort = attr.ib(kw_only=True, default=None, converter=_parse_custom_sort_properties)
```

Reply:
Something like that yes. You could even create a dataclass for the config instead of just using a dict

Reply:
https://github.com/dimagi/commcare-hq/pull/33621/commits/7b50e4bb195b6c1bce6be3d4d619e365cfd63707